libcontainer/cgroups/systemd/v1.go

```diff
@@ -326,6 +332,57 @@ func (m *legacyManager) GetStats() (*cgroups.Stats, error) {
 	return stats, nil
 }
 
+// freezeBeforeSet answers whether there is a need to freeze the cgroup before
+// applying its systemd unit properties, and thaw after, while avoiding
+// unnecessary freezer state changes.
+//
+// The reason why we have to freeze is that systemd's application of device
+// rules is done disruptively, resulting in spurious errors to common devices
+// (unlike our fs driver, they will happily write deny-all rules to running
+// containers). So we have to freeze the container to avoid the container get
+// an occasional "permission denied" error.
+func (m *legacyManager) freezeBeforeSet(unitName string, r *configs.Resources) (needsFreeze, needsThaw bool, Err error) {
```

Comment:
I think it would be nicer to call the returned error `Err` and have a special name for the error we intentionally skip returning in `SkipDevices` rather than the other way around.

Reply:
Yes, it is intentional -- I have decided to play it safe and not treat an error while getting a property value as fatal for Set, but rather just fall back to do the freeze. Not sure if that makes sense or not, just did not wanted to introduce another way for Set to fail.

Reply:
fixed `Err -> err`, added a comment about not returning an error from `getUnitProperty`.